src/sentry/auth/idpmigration.py

```diff
@@ -15,8 +37,19 @@ def create_verification_key(user: User, org: Organization, email: str) -> None:
     :param org: the organization whose SSO provider is being used
     :param email: the email address associated with the SSO identity
     """
-    key = str(uuid.uuid4())  # noqa
-    raise NotImplementedError  # TODO
+    redis_key = "verificationKeyStorage"
+    TTL = 60 * 10
+    cluster = redis.clusters.get("default").get_local_client_for_key(redis_key)
+    member_id = OrganizationMember.objects.get(organization=org, user=user).id
+
+    verification_code = default_validation_hash()
+    verification_key = f"auth:one-time-key:{verification_code}"
+    verification_value = {"user_id": user.id, "email": email, "member_id": member_id}
+
+    cluster.hmset(verification_key, verification_value)
+    cluster.expire(verification_key, TTL)
```

Comment:
Optional style advice: I'm always a fan of using objects over primitives where it adds meaning. In my opinion, it would improve the declaration to change it from `TTL = 60 * 10` to `TTL = timedelta(minutes=10)`. Then, you would just have to unpack it back to a primitive here, by
```python
cluster.expire(verification_key, TTL.total_seconds())
```

Reply:
Redis doesn't seem to like TTL.total_seconds() since it will return 600.0, so i used TTL.seconds instead

Reply:
`TTL.seconds` works in this case but I think `int(TTL.total_seconds())` is better style. The `seconds` attribute is just the seconds *part* of the timedelta being divided into days, seconds, and microseconds. (Observe that `timedelta(days=1).seconds` is 0.) Not much risk of a bug unless the TTL is adjusted to more than 24 hours, but better safe than sorry. 😉